src/lib/schemas.js

```diff
@@ -3,12 +3,12 @@ import { z } from 'zod';
 export const componentsSchema = z.array(
 	z.object({
 		title: z.string(),
-		npm: z
-			.string()
-			.regex(/(@[\w-]+\/)?[\w-]+/)
-			.optional(),
+		npm: z.string().regex(/^(@[a-z0-9-]*\/)?[a-z0-9-]*$/),
 		url: z.string().url().optional(),
-		repository: z.string().url(),
+		repository: z
+			.string()
+			.url()
+			.refine((val) => val.includes('github.com') || val.includes('gitlab.com')),
```

Comment:
I don't want to restrict repositories to only GitLab or GitHub.

Someone could want to host the code on Bitbucket, Sourceforge, etc.
We don't want to prevent them to add the component.

Today we don't have those cases, and so the script that update stars don't handle repositories others than GitHub or GitLab, but if there are others repository we will update the script

Reply:
Good point!